### DISCUSSION TITLE
Adventure map, Sign and Bottle without text should randomly display predefined texts

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#10305] [General]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

In fheroes2, the Sign and Bottle without text should randomly display predefined texts. **[Done]**
In original game, these texts are displayed when it is the case:

"Next sign 50 miles"
"Burma shave"
"See Rock City"
"This space for rent"

**Edit.**

Random texts should be assigned at the beginning of the scenario so that when a player visits the same sign several times, the same text is displayed.
The same goes for bottles, if a player visits a bottle, and loads a previous save, the same text should be displayed.

Note: the yellow titles "Sign" and "Bottle" must be kept in fheroes2.

**Suggestions.**

Question: should we put a dot "." at the end of these texts (to remain consistent with the texts of the other game windows)?
I am in favor of this.

Also, the text "This space for rent" is not really fitting for the Bottle because the Bottle disappears after being visited.
What do you think about this to replace it:
(From the original "Mission: Impossible" TV series.)

"This message will self-destruct in five seconds. Good luck Jim."

Or just:
(If you think it would be necessary in order to eventually avoid copyrights.)

"This message will self-destruct in five seconds."

I also suggest, for the Bottle, to remplace the text by:

"Next bottle 50 miles"

Instead of:

"Next sign 50 miles"

In original game:

![Sign bottle text 001](https://user-images.githubusercontent.com/43583994/140921794-a57cddbe-0395-4611-8a6a-46f2fb7b8e8f.png)
![Sign bottle text 002](https://user-images.githubusercontent.com/43583994/140921804-00fe492a-2f41-4614-8991-d2b93427aae3.png)
![Sign bottle text 003](https://user-images.githubusercontent.com/43583994/140921819-dadfac6b-9597-4787-b46a-d53546351172.png)
![Sign bottle text 004](https://user-images.githubusercontent.com/43583994/140921828-44505a17-1abf-4113-aa66-2b98b7cb9c4a.png)
![Sign bottle text 005](https://user-images.githubusercontent.com/43583994/140921837-a1aa0347-cc60-4d67-a6c7-455e82a333e8.png)
![Sign bottle text 006](https://user-images.githubusercontent.com/43583994/140921848-ab98af5a-7158-463a-8842-954baa0e5249.png)
![Sign bottle text 007](https://user-images.githubusercontent.com/43583994/140921851-c546fc88-a533-4fff-9e87-c8802a92f6fe.png)
![Sign bottle text 008](https://user-images.githubusercontent.com/43583994/140921863-82b0ebad-a9d6-49fd-a508-949164dab15f.png)

In fheroes2:

![Sign bottle text 009](https://user-images.githubusercontent.com/43583994/140922106-7099c881-542b-4a3a-9156-dd40a9c30e03.png)
![Sign bottle text 010](https://user-images.githubusercontent.com/43583994/140922120-edb7097d-997b-406f-86d7-e3947044f138.png)

### Save file

Here are the save files:
[Before signs and bottles 01.zip](https://github.com/ihhub/fheroes2/files/7504559/Before.signs.and.bottles.01.zip)

**Comment by pflugs30** [comment]
I continue to be amazed at how thorough you are in your testing. 😀  Good find!

**Comment by ihhub** [comment]
Haha, burma shave...

**Comment by pflugs30** [comment]
@zenseii more translations coming... 😉

**Comment by LeHerosInconnu** [comment]
Hello @pflugs30,

> I continue to be amazed at how thorough you are in your testing. :grinning: Good find!

Ah, that's also why there are still over 400 issues. :rofl:

**Comment by ihhub** [comment]
> 
> 
> Hello @pflugs30,
> 
> > I continue to be amazed at how thorough you are in your testing. 😀 Good find!
> 
> Ah, that's also why there are still over 400 issues. 🤣

and this disturbs me all the time...

**Comment by LeHerosInconnu** [comment]
Hello @ihhub,

> > Hello @pflugs30,
> > > I continue to be amazed at how thorough you are in your testing. grinning Good find!
> > 
> > 
> > Ah, that's also why there are still over 400 issues. :rofl:
> 
> and this disturbs me all the time...

"I don't understand.
The more issues I correct, the more there are." :rofl:

**Comment by zenseii** [comment]
Should these be considered the same issue as these missing strings for when you pick up an artifact?
![image](https://user-images.githubusercontent.com/12501091/141447399-bc4b51ef-7029-44cf-b2b1-688630ac0d5c.png)

In the map editor this telescope is a random artifact. If I load and return to this spot I will always get this message.

**Comment by LeHerosInconnu** [comment]
Hello @zenseii,

> Should these be considered the same issue as these missing strings for when you pick up an artifact?
> ![image](https://user-images.githubusercontent.com/12501091/141447399-bc4b51ef-7029-44cf-b2b1-688630ac0d5c.png)
> 
> In the map editor this telescope is a random artifact. If I load and return to this spot I will always get this message.

When artifacts are set to random in the scenario editor, this means that at the beginning of the scenario all these artifacts are set.
After that, if a player loads a save game that he made as soon as he got the hand (or any time later in the game), all these random artifacts will always be the same.

Now, if a player restarts a scenario completely (new game), the random artifacts are set at the beginning of the scenario and are different from another game of the same scenario.

**Comment by ihhub** [comment]
Hi @LeHerosInconnu  , I've added the original behavior of signs and bottles. Could you please update your first post to indicate the suggestions over the original game?

**Comment by LeHerosInconnu** [comment]
Hello @ihhub,

> Hi @LeHerosInconnu , I've added the original behavior of signs and bottles. Could you please update your first post to indicate the suggestions over the original game?

Done.

**Comment by ihhub** [comment]
The original issue had been fixed. We need to agree on new changes for texts.
